### PR TITLE
helper/schema: generate strings for ValueType

### PR DESCRIPTION
**Pull request by mitchellh**
This uses the new `go generate` in Go 1.4 to generate string values for our enums.

/cc @armon @ryanuber @pearkes

### CODE REVIEW

File: helper/schema/valuetype_string.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// generated by stringer -type=ValueType; DO NOT EDIT
+
+package schema
+
+import "fmt"
+
+const _ValueType_name = "TypeInvalidTypeBoolTypeIntTypeStringTypeListTypeMapTypeSettypeObject"
```
Comment (miksago) [inline helper/schema/valuetype_string.go]: Out of curiosity, is the last `type` in this correct, being lowercased rather than uppercased like the rest?

Comment (mitchellh) [inline helper/schema/valuetype_string.go]: Yeah, at some point we might want to capitalize it but that use case isn't well tested to expose it as a public API and we needed it internally for private APIs.

Comment (miksago) [inline helper/schema/valuetype_string.go]: Ah, okay, just looked like an accidental error.

Comment (mitchellh) [inline helper/schema/valuetype_string.go]: Yeah, I appreciate the double check :)

**Pull request merged**